synapse/config/oembed.py

```diff
@@ -93,11 +95,22 @@ def _parse_and_validate_provider(
             # might have multiple patterns to match.
             for endpoint in provider["endpoints"]:
                 api_endpoint = endpoint["url"]
+
+                # The API endpoint must be an HTTP(S) URL.
+                results = urlparse.urlparse(api_endpoint)
+                if results.scheme not in {"http", "https"}:
+                    raise ConfigError(
+                        f"Insecure oEmbed scheme ({results.scheme}) for endpoint {api_endpoint}",
```

Comment:
```suggestion
                        f"Unsupported oEmbed scheme ({results.scheme}) for endpoint {api_endpoint}",
```

Otherwise I assumed it meant we were forbidding `http` schemes